src/stream/reader/logfile.rs

```diff
@@ -0,0 +1,373 @@
+use crate::client::commands::log::{CatOpts, Channel, ShowOpts};
+use crate::transfer::stream::ChannelId;
+use crate::{JobTaskCount, JobTaskId, Map};
+use byteorder::ReadBytesExt;
+use colored::{Color, Colorize};
+use std::collections::BTreeMap;
+use std::fs::File;
+use std::io::{BufReader, ErrorKind, Read, Seek, SeekFrom, Write};
+use std::path::Path;
+use tako::InstanceId;
+
+pub const BLOCK_STREAM_START: u8 = 0;
+pub const BLOCK_STREAM_CHUNK: u8 = 1;
+pub const BLOCK_STREAM_END: u8 = 2;
+
+pub struct ChunkInfo {
+    position: u64,
+    size: u32, // Currently chunk is actually limited to 128kB
+}
+
+pub struct InstanceInfo {
+    channels: [Vec<ChunkInfo>; 2],
+    finished: bool,
+}
+
+pub struct TaskInfo {
+    instances: Map<InstanceId, InstanceInfo>,
```

Comment:
Since usually there will be only one instance and they will be mostly contiguous, maybe `(Small)Vec<(InstanceId, InstanceInfo>)` could take less memory?

Reply:
Yes, you are totally right. But I was trying to avoid some (premature) optimizations to make working version ASAP.